src/Network/HTTP/Download.hs

```diff
@@ -108,7 +108,7 @@ redownload req0 dest = do
                         requestHeaders req0 ++
                         [("If-None-Match", L.toStrict etag)]
                     }
-        req2 = req1 { checkStatus = \_ _ _ -> Nothing }
+        req2 = req1 { checkResponse = \_ _ -> return () }
```

Comment:
@borsboom: In this case the we're handling the status codes manually afterwards, so I think it's still good to override the `checkResponse` method that's passed in.